packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -61,6 +62,45 @@ import {
 	createOverlappingIntervalsIndex,
 } from "./intervalIndex";
 
+/**
+ * Defines a position and side relative to a character in a sequence.
+ *
+ * For this purpose, sequences look like:
+ *
+ * `{start} - {character 0} - {character 1} - ... - {character N} - {end}`
+ *
+ * Each `{value}` in the diagram is a character within a sequence.
+ * Each `-` in the above diagram is a position where text could be inserted.
+ * Each position between a `{value}` and a `-` is a `SequencePlace`.
+ *
+ * The special endpoints `{start}` and `{end}` refer to positions outside the
+ * contents of the string.
+ *
+ * This gives us 2N + 2 possible positions to refer to within a string, where N
+ * is the number of characters.
+ */
+export type SequencePlace = number | "start" | "end" | InteriorSequencePlace;
```

Comment:
add a comment documenting that `number` is equivalent to specifying `InteriorSequencePlace` with `side: Side.Before`